scripts/state/collision/collision.js

```diff
@@ -0,0 +1,60 @@
+class Collision {
+
+  collisions(allBullets, allShips, sounds) {
+    let bulletsVis = [];
```

Comment:
ok, most of the bullets will be visible (like 80% to 90%) with the system so we really don't gain much from this, and this is just iterating over them an extra time. but its fine.